src/reporter/index.ts

```diff
@@ -137,6 +142,26 @@ export default class Reporter {
         return promise;
     }
 
+    public async init (): Promise<ReporterInitMethodResult> {
+        let error = void 0;
+
+        try {
+            await this.dispatchToPlugin({
```

Comment:
The whole purpose of the `dispatchToPlugin` method is handling errors coming from plugin methods. I feel that doing additional error handling here is redudant.

First, maybe it's better to use a generalized error message like
```
The "xxx" reporter method "yyy" has thrown an error:
```
and use this wrapper for all kind of errors in `dispatchToPlugin`

Second, you can just use `this.messageBus` as the `initialObject` argument of `dispatchToPlugin`. This way it will be passsed through the message bus directly to the runner and tests will be terminated by the runner itself. It can reduce the boilerplate code passing the error around different methods.